CSharp/ArticleProjects/TaskDependencies/Program.cs

```diff
@@ -58,105 +56,188 @@ public static void Main(string[] args)
+                        new CloudTask("Sun", "cmd.exe /c echo Sun"),
  
-                    // Tasks 1, 2, and 3 don't depend on any other tasks. Because
-                    // we will be using them for a task range dependency, we must
-                    // specify string representations of integers as their ids.
-                    new CloudTask("1", "cmd.exe /c echo 1"),
-                    new CloudTask("2", "cmd.exe /c echo 2"),
-                    new CloudTask("3", "cmd.exe /c echo 3"),
+                        // Task 'Flowers' depends on completion of both 'Rain' and 'Sun'
+                        // before it is run.
+                        new CloudTask("Flowers", "cmd.exe /c echo Flowers")
+                        {
+                            DependsOn = TaskDependencies.OnIds("Rain", "Sun")
+                        },
  
-                    // Task 4 depends on a range of tasks, 1 through 3
-                    new CloudTask("4", "cmd.exe /c echo 4")
+                        // Tasks 1, 2, and 3 don't depend on any other tasks. Because
+                        // we will be using them for a task range dependency, we must
+                        // specify string representations of integers as their ids.
+                        new CloudTask("1", "cmd.exe /c echo 1"),
+                        new CloudTask("2", "cmd.exe /c echo 2"),
+                        new CloudTask("3", "cmd.exe /c echo 3"),
+ 
+                        // Task 4 depends on a range of tasks, 1 through 3
+                        new CloudTask("4", "cmd.exe /c echo 4")
+                        {
+                            // To use a range of tasks, their ids must be integer values.
+                            // Note that we pass integers as parameters to TaskIdRange,
+                            // but their ids (above) are string representations of the ids.
+                            DependsOn = TaskDependencies.OnIdRange(1, 3)
+                        },
+
+                        // Task 5 depends on a range of tasks, 1 through 3, and 'Flowers'
+                        new CloudTask("5", "cmd.exe /c echo 5")
+                        {
+                            DependsOn = new TaskDependencies(
+                                new[] { "Flowers" },
+                                new[] { new TaskIdRange(1, 3) })
+                        },
+
+                        // Task A is a parent task that exits with a non-zero exit code.
+                        new CloudTask("A", "cmd.exe /c echo A exit 1"),
+                        // Task B depends on task A
+                        new CloudTask("B", "cmd.exe /c echo B")
+                        {
+                            DependsOn = TaskDependencies.OnId("A"),
+                            ExitConditions = new ExitConditions()
+                            {
+                                // If task A exits with a non-zero exit code, run downstream task B.
```

Comment:
To have an exit option for non-zero exit codes *only*, you would need to either:

1. Use ExitCodeRanges to specify a range that takes in all possible codes, e.g. 1 to 2147483647

2. Set Default to the desired 'all non-zero exit codes' behaviour and have an explicit SchedulingError clause to override it for the scheduling error case